packages/core/core/src/migrations/database/5.0.0-discard-drafts.ts

```diff
@@ -113,7 +113,7 @@ export async function* getBatchToDiscard({
   db,
   trx,
   uid,
-  batchSize = 1000,
+  batchSize = 500,
 }: {
   db: Database;
   trx: Knex;
```

Comment:
Could I suggest we do this then? wdyt @Convly? (suggestion for the entire function)
```suggestion
/**
 * Load a batch of versions to discard.
 *
 * Versions with only a draft version will be ignored.
 * Only versions with a published version (which always have a draft version) will be discarded.
 */
export async function* getBatchToDiscard({
  db,
  trx,
  uid,
  defaultBatchSize = 1000,
}: {
  db: Database;
  trx: Knex;
  uid: string;
  defaultBatchSize?: number;
}) {
  const client = db.config.connection.client;
  const isSQLite = ['sqlite', 'sqlite3', 'better-sqlite3'].includes(client as any);

  const batchSize = isSQLite ? 500 : defaultBatchSize;

  let offset = 0;
  let hasMore = true;

  while (hasMore) {
    // Look for the published entries to discard
    const batch: DocumentVersion[] = await db
      .queryBuilder(uid)
      .select(['id', 'documentId', 'locale'])
      .where({ publishedAt: { $ne: null } })
      .limit(batchSize)
      .offset(offset)
      .orderBy('id')
      .transacting(trx)
      .execute();

    if (batch.length < batchSize) {
      hasMore = false;
    }

    offset += batchSize;
    yield batch;
  }
}
```

Reply:
If we can get rid of the any + add a small explanation, it looks good to me

For example, we can reference https://www.sqlite.org/limits.html (7) or directly quote an extract of

```
Maximum Number Of Terms In A Compound SELECT Statement

A compound  SELECT statement is two or more SELECT statements connected by operators UNION, UNION ALL, EXCEPT, or INTERSECT. We call each individual SELECT statement within a compound SELECT a "term".

The code generator in SQLite processes compound SELECT statements using a recursive algorithm. In order to limit the size of the stack, we therefore limit the number of terms in a compound SELECT. The maximum number of terms is SQLITE_MAX_COMPOUND_SELECT which defaults to 500. We think this is a generous allotment since in practice we almost never see the number of terms in a compound select exceed single digits.
```